doc/topics/the-event-loop-timers-and-nexttick.md

```diff
@@ -0,0 +1,375 @@
+of time depends on how many callbacks end up being executed and how long they
+take._
+
+For example, say you schedule a timeout to execute after a 100 ms threshold,
+then your script starts asynchronously reading a file which takes 95 ms:
+
+```js
+
+var fs = require('fs');
+
+function someAsyncOperation (callback) {
+  
+  // let's assume this takes 95ms to complete  
+  fs.readFile('/path/to/file', callback);
+
+}
+
+var timeoutScheduled = Date.now();
+
+setTimeout(function () {
+
+  var delay = Date.now() - timeoutScheduled;
+
+  console.log(delay + "ms have passed since I was scheduled");
+}, 100);
+
+
+// do someAsyncOperation which takes 95 ms to complete
+someAsyncOperation(function () {
+
+  var startCallback = Date.now();
+
+  // do something that will take 10ms...
+  while (Date.now() - startCallback < 10) {
+    ; // do nothing
+  }
+
+});
+```
+
+When the event loop enters the `poll` phase, it has an empty queue
+(`fs.readFile()` has not completed) so it will wait for the number of ms
+remaining until the soonest timer's threshold is reached.  While it is
+waiting 95 ms pass, `fs.readFile()` finishes reading the file and its
+callback which takes 10 ms to complete is added to the `poll` queue and
+executed.  When the callback finishes, there are no more callbacks in the
+queue, so the event loop will see  that the threshold of the soonest timer has
+been reached then wrap back to the `timers` phase to execute the  timer's
+callback.  In this example, you will see that the total delay between
+the timer being scheduled and its callback being executed will be
+105ms.
+
+Note: To prevent the `poll` phase from starving the event loop, libuv also has
+a hard maximum (system dependent) before it stops poll'ing for more events.
+
+### pending callbacks:
+
+This phase executes callbacks for some system operations such as types of TCP
+errors.  For example if a TCP socket receives `ECONNREFUSED` when
+attempting to connect, some *nix systems want to wait to report the
+error. This will be queued to execute in the `pending callbacks` phase.
+
+### poll:  
+
+When the event loop enters the `poll` phase _and there are no timers
+scheduled_, one of two things will happend: 
+
+* _if the `poll` queue **is not empty**_, the event loop will iterate through 
+its queue of callbacks executing them synchronously until either the queue has 
+been exhausted, or the system-dependent hard limit is reached. 
+
+* _if the `poll` queue is **empty**_, the event loop will wait for a
+callback
+to be added to the queue, then execute it immediately. 
+
+Once the `poll` queue is empty the event loop will check for timers _whose
+time thresholds have been reached_.  If one or more timers are ready, the
+event loop will wrap back to the timers phase to execute  those timers'
+callbacks.
+
+### `setImmediate()`:
+
+`setImmediate()` is actually a special timer that runs in a separate phase of
```

Comment:
I'm not sure that I would call it a timer really since it has no "timeout" value per se ...